app/server.go

```diff
@@ -150,11 +142,20 @@ func NewServer(options ...Option) (*Server, error) {
 		}
 	}
 
-	if err := s.LoadConfig(s.configFile); err != nil {
-		return nil, err
-	}
+	if s.configStore == nil {
+		configStore, needsSave, err := config.NewFileStore("config.json", true)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to load config")
+		}
 
-	s.EnableConfigWatch()
+		s.configStore = configStore
+		if needsSave {
```

Comment:
I'm not super happy about `needsSave`, but our current config loading model involves detecting missing settings or warning and "fixing" them without erroring out altogether. I really didn't want `Load()` (invoked through `NewFileStore`) to have implicit write semantics, so I've forced that on the caller instead to do the requisite `Save`. This, of course, conflicts with my desire to eliminate `Save` altogether and make it implicit. Perhaps it needs to stay...